pkg/controller.v1/pytorch/envvar.go

```diff
@@ -90,16 +90,19 @@ func setPodEnv(obj interface{}, podTemplateSpec *corev1.PodTemplateSpec, rtype,
 				Name:  "RANK",
 				Value: strconv.Itoa(rank),
 			})
-			podTemplateSpec.Spec.Containers[i].Env = append(podTemplateSpec.Spec.Containers[i].Env, corev1.EnvVar{
-				Name:  EnvNprocPerNode,
-				Value: *pytorchjob.Spec.NprocPerNode,
-			})
 			podTemplateSpec.Spec.Containers[i].Env = append(podTemplateSpec.Spec.Containers[i].Env, corev1.EnvVar{
 				Name:  EnvNodeRank,
 				Value: strconv.Itoa(rank),
 			})
 		}
 
+		if pytorchjob.Spec.NprocPerNode != nil {
+			podTemplateSpec.Spec.Containers[i].Env = append(podTemplateSpec.Spec.Containers[i].Env, corev1.EnvVar{
+				Name:  EnvNprocPerNode,
+				Value: *pytorchjob.Spec.NprocPerNode,
+			})
+		}
+
```

Comment:
this change removes the necessity to add master spec to run multiple processes per pod using python/torchrun launch method without the elastic spec in the yaml, but doesn't solve #1947

Reply:
to solve #1947 I think we need to make changes here https://github.com/kubeflow/training-operator/blob/230bfb453188ecc0e9def4296e75f4bd06707769/pkg/controller.v1/pytorch/elastic.go#L102

Reply:
@deepanker13 elastic.go has env vars when elastic policy is set. As per the original issue, it works correctly when `elasticPolicy.NProcPerNode` is set.

Reply:
@johnugeorge following this pr https://github.com/kubeflow/training-operator/pull/1840 , I raised issue #1947  which should complete the point striked out as attached in the below screenshot
<img width="938" alt="Screenshot 2023-11-20 at 11 37 10 AM" src="https://github.com/kubeflow/training-operator/assets/121787544/bc36ac19-12c2-4986-9f9f-d473197c732c">

Reply:
> to solve #1947 I think we need to make changes here
> 
> https://github.com/kubeflow/training-operator/blob/230bfb453188ecc0e9def4296e75f4bd06707769/pkg/controller.v1/pytorch/elastic.go#L102

No. this part will be deprecated. We should touch the global one.